src/sparse/impl/KokkosSparse_spmv_impl.hpp

```diff
@@ -908,59 +1225,62 @@ spmv_alpha_beta_mv_no_transpose (const typename YVector::non_const_value_type& a
     // Assuming that no row contains duplicate entries, NNZPerRow
     // cannot be more than the number of columns of the matrix.  Thus,
     // the appropriate type is ordinal_type.
-    const ordinal_type NNZPerRow = static_cast<ordinal_type> (A.nnz () / A.numRows ());
-
-    int vector_length = 1;
-    while( (static_cast<ordinal_type> (vector_length*2*3) <= NNZPerRow) && (vector_length<8) ) vector_length*=2;
+    const ordinal_type numRows = A.numRows();
+    const ordinal_type NNZPerRow = static_cast<ordinal_type> (A.nnz () / numRows);
 
 #ifndef KOKKOS_FAST_COMPILE // This uses templated functions on doalpha and dobeta and will produce 16 kernels
```

Comment:
Do we ever test with `KOKKOS_FAST_COMPILE`?  It's a noble idea, but useless if not tested regularly.  We should just make the one set of kernels build in reasonable time.